### PR TITLE
Channel name testing suggestions

### PR DESCRIPTION
**Pull request by CodyCBakerPhD**
## Motivation

@h-mayorquin Here are those suggestions I said I'd send your way with regards to improving the design of the new test class for appending electrode tables, and eventually what all those SI tests ought to look like for optimal performance and readability.

Let me know if you have any questions about anything.

## Checklist

- [X] Have you checked our [Contributing](https://github.com/catalystneuro/nwb-conversion-tools/blob/master/docs/contribute.rst) document?
- [X] Have you ensured the PR description clearly describes the problem and solutions?
- [X] Have you checked to ensure that there aren't other open or previously closed [Pull Requests](https://github.com/catalystneuro/nwb-conversion-tools/pulls) for the same change?
- [X] If this PR fixes an issue, is the first line of the PR description `fix #XXX` where `XXX` is the issue number?

### CODE REVIEW

File: tests/test_internals/test_si.py [hunk 1]
```diff
@@ -585,66 +585,56 @@ def test_group_set_custom_description(self):
                     assert nwb.electrodes["group"][i].description == "M1 description"
 
 
-class TestSpikeInterfaceRecorders(unittest.TestCase):
+class TestAddElectrodes(unittest.TestCase):
```
Comment (h-mayorquin) [inline tests/test_internals/test_si.py:588]: I think this is way more descriptive of what the test is doing right now. I was thinking that we would like to add more `spikeinterface` specific functionality that we would like to test in the future and decided to leave it more generally. Maybe this is an instance of early optimization (early generalization?). Note also that there is a test called `TestWriteElectrodes` above leading to some semantic overlap.

File: tests/test_internals/test_si.py [hunk 2]
```diff
@@ -585,66 +585,56 @@ def test_group_set_custom_description(self):
...
+    def test_channel_names(self):
+        """Ensure channel names merge correctly after appending."""
+        add_electrodes(recording=self.recording_1, nwbfile=self.nwbfile)
+        add_electrodes(recording=self.recording_2, nwbfile=self.nwbfile)
 
-        channel_ids = cls.RX1.get_channel_ids()
-        num_channels = cls.RX1.get_num_channels()
-        non_int_channel_ids1 = ["a", "b", "c", "d"]
-        non_int_channel_ids2 = ["c", "d", "e", "f"]
+        self.assertListEqual(list(self.nwbfile.electrodes["channel_name"].data), ["a", "b", "c", "d", "e", "f"])
```
Comment (h-mayorquin) [inline tests/test_internals/test_si.py:613]: Out of curiosity, what about casting the `["a", "b", ...]` to array and use numpy testing tools?

File: tests/test_internals/test_si.py [hunk 3]
```diff
@@ -585,66 +585,56 @@ def test_group_set_custom_description(self):
...
-        cls.RX_non_int_channels1 = cls.RX1.channel_slice(
-            channel_ids=channel_ids, renamed_channel_ids=non_int_channel_ids1
-        )
-        cls.RX_non_int_channels2 = cls.RX1.channel_slice(
-            channel_ids=channel_ids, renamed_channel_ids=non_int_channel_ids2
-        )
+    def test_common_property_extension(self):
+        """Add a property for a first recording that is then extended by a second recording."""
+        self.recording_1.set_property(key="property", values=["value_1"] * self.num_channels)
+        self.recording_2.set_property(key="property", values=["value_2"] * self.num_channels)
```
Comment (h-mayorquin) [inline tests/test_internals/test_si.py:618]: My leaning here is to change "property" to "common_property" and "property2" to "added_property" for even more expressiveness. Likewise, I would have the following map for their values:
* "value_1" -> "common_property_value_in_recording_1"
* "value_2" -> "common_property_value_in_recording_2"
* "value_3" -> "value_of_added_property"  

I do think that we are discussing aesthetics at this level, or what do you think? can you cut this one with arguments for one side or the other?

**Review by h-mayorquin — COMMENTED** [review]
Hey, thanks a lot for doing this. With this I have a better idea of what makes sense or how does a good unittesting looks like. Points to note for me are:
* You re-initialize the nwbfile. I guess this provides a cleaner file while debugging which isolates what is tested in line with the philosophy of unitesting. I guess this is also the reason why you slice the channels again in order to erase their properties? 
* I like the names that you chose for the unitest better. I guess I was hanged up in the fact that we were testing non int channel ids that did not think in the more general properties that we are testing (which implicity test for non int channels) and the resulting names I had were and felt too cumbersome. 
* Documentation of unnitest by giving a brief comment of the intent.

Also, by the end the `TearDown` method for cleaning the directory was not necessary?

**Comment by h-mayorquin** [comment]
Also, I prefer the specificity of `recorder` to the RX convention that we use here and in other places of our library (e.g. as a class attribute of the data interfaces).

File: tests/test_internals/test_si.py (hunk 3 above)
Comment (h-mayorquin) [reply tests/test_internals/test_si.py:618]: I will take care of this now, then.

**Comment by CodyCBakerPhD** [comment]
> Also, I prefer the specificity of `recorder` to the RX convention that we use here and in other places of our library (e.g. as a class attribute of the data interfaces).

Exactly right; the original issue with that (and still present elsewhere in the file) is a name like `RX` implies that its a **class**, not an **instance** of the class. Instances should be all lowercase with underscore separation whereas classes generally follow camel case.

**Pull request merged**